diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd
index 47911da..5bbaae2 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_2G.dtd
@@ -274,7 +274,7 @@ void root_func(){
 		dev_name = "mesonvout";
 		status = "okay";
 	};
-/*
+
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V Amvideocap
 //$$ MODULE = "DISP&MM-Amvideocap"
@@ -288,7 +288,7 @@ void root_func(){
 		reserve-memory = <0x00600000>;//1920*1088*3
 		reserve-iomap = "true";
 	};
-*/
+
 /// ***************************************************************************************
 ///	-	RTC
 //$$ MODULE="RTC"
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd b/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd
index 39359a8..08a94dd 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200a_2G_4k2k.dtd
@@ -274,7 +274,7 @@ void root_func(){
 		dev_name = "mesonvout";
 		status = "okay";
 	};
-/*
+
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V Amvideocap
 //$$ MODULE = "DISP&MM-Amvideocap"
@@ -288,7 +288,7 @@ void root_func(){
 		reserve-memory = <0x00600000>;//1920*1088*3
 		reserve-iomap = "true";
 	};
-*/
+
 /// ***************************************************************************************
 ///	-	RTC
 //$$ MODULE="RTC"
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
index 6d383b9..310d3e2 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
@@ -297,7 +297,7 @@ void root_func(){
 		dev_name = "mesonvout";
 		status = "okay";
 	};
-/*
+
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V Amvideocap
 //$$ MODULE = "DISP&MM-Amvideocap"
@@ -311,7 +311,7 @@ void root_func(){
 		reserve-memory = <0x00600000>;//1920*1088*3
 		reserve-iomap = "true";
 	};
-*/
+
 /// ***************************************************************************************
 ///	-	RTC
 //$$ MODULE="RTC"
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
index fcb5e03..7601f23 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
@@ -297,7 +297,7 @@ void root_func(){
 		dev_name = "mesonvout";
 		status = "okay";
 	};
-/*
+
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V Amvideocap
 //$$ MODULE = "DISP&MM-Amvideocap"
@@ -311,7 +311,7 @@ void root_func(){
 		reserve-memory = <0x00600000>;//1920*1088*3
 		reserve-iomap = "true";
 	};
-*/
+
 /// ***************************************************************************************
 ///	-	RTC
 //$$ MODULE="RTC"
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
index 484ac36..765cdec 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
@@ -274,7 +274,7 @@ void root_func(){
 		dev_name = "mesonvout";
 		status = "okay";
 	};
-/*
+
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V Amvideocap
 //$$ MODULE = "DISP&MM-Amvideocap"
@@ -288,7 +288,7 @@ void root_func(){
 		reserve-memory = <0x00800000>;//1920*1088*4
 		reserve-iomap = "true";
 	};
-*/
+
 /// ***************************************************************************************
 ///	-	RTC
 //$$ MODULE="RTC"
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
index 7f1231b..461eb30 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
@@ -274,7 +274,7 @@ void root_func(){
 		dev_name = "mesonvout";
 		status = "okay";
 	};
-/*
+
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V Amvideocap
 //$$ MODULE = "DISP&MM-Amvideocap"
@@ -288,7 +288,7 @@ void root_func(){
 		reserve-memory = <0x00600000>;//1920*1088*3
 		reserve-iomap = "true";
 	};
-*/
+
 /// ***************************************************************************************
 ///	-	RTC
 //$$ MODULE="RTC"
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd b/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd
index 16b6b69..72f0a7a 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd
@@ -312,6 +312,20 @@ void root_func(){
 	};
 
 /// ***************************************************************************************
+///	-	DISP&MM-A/V Amvideocap
+//$$ MODULE = "DISP&MM-Amvideocap"
+//$$ DEVICE="amvideocap"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 4 ="reg"
+	amvideocap{
+		compatible = "amlogic,amvideocap";
+		dev_name = "amvideocap.0";
+		status = "okay";
+		reserve-memory = <0x00600000>;//1920*1088*3
+		reserve-iomap = "true";
+	};
+
+/// ***************************************************************************************
 ///	-	UART
 //$$ MODULE="UART"
 //$$ DEVICE="uart_ao"
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd
index dd00abb..6b1efb8 100755
--- a/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd
@@ -300,7 +300,7 @@ void root_func(){
 		dev_name = "mesonvout";
 		status = "okay";
 	};
-/*
+
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V Amvideocap
 //$$ MODULE = "DISP&MM-Amvideocap"
@@ -314,7 +314,7 @@ void root_func(){
 		reserve-memory = <0x00600000>;//1920*1088*3
 		reserve-iomap = "true";
 	};
-*/
+
 /// ***************************************************************************************
 ///	-	RTC
 //$$ MODULE="RTC"
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
index b711f7d..b0d6ee8 100755
--- a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
@@ -292,7 +292,7 @@ void root_func(){
 		dev_name = "mesonvout";
 		status = "okay";
 	};
-/*
+
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V Amvideocap
 //$$ MODULE = "DISP&MM-Amvideocap"
@@ -306,7 +306,7 @@ void root_func(){
 		reserve-memory = <0x00600000>;//1920*1088*3
 		reserve-iomap = "true";
 	};
-*/
+
 /// ***************************************************************************************
 ///	-	RTC
 //$$ MODULE="RTC"
