diff -Nru linux-amlogic-3.10.40.orig/arch/arm/mach-meson8b/pm.c linux-amlogic-3.10.40/arch/arm/mach-meson8b/pm.c
--- linux-amlogic-3.10.40.orig/arch/arm/mach-meson8b/pm.c	2014-11-24 04:54:32.000000000 +0300
+++ linux-amlogic-3.10.40/arch/arm/mach-meson8b/pm.c	2014-12-04 12:16:28.409151522 +0300
@@ -10,6 +10,7 @@
 
 #include <linux/pm.h>
 #include <linux/suspend.h>
+#include <linux/reboot.h>
 #include <linux/module.h>
 #include <linux/platform_device.h>
 #include <linux/clk.h>
@@ -346,6 +347,8 @@
 static void meson_pm_finish(void)
 {
     printk(KERN_INFO "enter meson_pm_finish!\n");
+    #include <asm/system.h>
+    machine_restart("");
 }
 
 static struct platform_suspend_ops meson_pm_ops = {
