diff -Nru linux-amlogic-3.10.40.orig/arch/arm/mach-meson8b/clock.c linux-amlogic-3.10.40/arch/arm/mach-meson8b/clock.c
--- linux-amlogic-3.10.40.orig/arch/arm/mach-meson8b/clock.c	2014-11-24 04:54:32.000000000 +0300
+++ linux-amlogic-3.10.40/arch/arm/mach-meson8b/clock.c	2014-12-02 10:58:56.752762721 +0300
@@ -1166,10 +1166,10 @@
 	cpu_clk_cntl = sys_pll_settings[idx][1];
 	latency.d32 =  sys_pll_settings[idx][2];
 
-	printk(KERN_DEBUG"CTS_CPU_CLK %4ld --> %4ld (MHz)\n",
-									clk->rate / 1000000, dst / 1000000);
-	pr_debug("CTS_CPU_CLK old_cntl=0x%x new_cntl=0x%x, latency: %x\n", 
-									curr_cntl, cpu_clk_cntl, latency.d32);
+	//printk(KERN_DEBUG"CTS_CPU_CLK %4ld --> %4ld (MHz)\n",
+	//								clk->rate / 1000000, dst / 1000000);
+	//pr_debug("CTS_CPU_CLK old_cntl=0x%x new_cntl=0x%x, latency: %x\n", 
+	//								curr_cntl, cpu_clk_cntl, latency.d32);
 
 	if (cpu_clk_cntl != curr_cntl) {
 SETPLL:
