diff --git a/arch/arm/mach-meson8/clock.c b/arch/arm/mach-meson8/clock.c
index cbd81cd..2d36da3 100755
--- a/arch/arm/mach-meson8/clock.c
+++ b/arch/arm/mach-meson8/clock.c
@@ -1153,8 +1153,8 @@ static int set_sys_pll(struct clk *clk,  unsigned long dst)
 
 	/*printk(KERN_DEBUG"CTS_CPU_CLK %4ld --> %4ld (MHz)\n",
 									clk->rate / 1000000, dst / 1000000);*/
-	pr_debug("CTS_CPU_CLK old_cntl=0x%x new_cntl=0x%x, latency: %x\n", 
-									curr_cntl, cpu_clk_cntl, latency.d32);
+	/*pr_debug("CTS_CPU_CLK old_cntl=0x%x new_cntl=0x%x, latency: %x\n", 
+									curr_cntl, cpu_clk_cntl, latency.d32);*/
 
 	if (cpu_clk_cntl != curr_cntl) {
 SETPLL:
