diff -Nru linux-amlogic-2058c9f.orig/arch/arm/plat-meson/power.c linux-amlogic-2058c9f/arch/arm/plat-meson/power.c
--- linux-amlogic-2058c9f.orig/arch/arm/plat-meson/power.c	2014-12-15 09:21:35.000000000 +0300
+++ linux-amlogic-2058c9f/arch/arm/plat-meson/power.c	2014-12-17 19:38:14.359627565 +0300
@@ -41,7 +41,9 @@
 {
     u32 reboot_reason = MESON_NORMAL_BOOT;
     if (cmd) {
-        if (strcmp(cmd, "charging_reboot") == 0)
+        if (strcmp(cmd, "poweroff") == 0)
+            reboot_reason = LINUX_REBOOT_CMD_POWER_OFF;
+        else if (strcmp(cmd, "charging_reboot") == 0)
             reboot_reason = MESON_CHARGING_REBOOT;
         else if (strcmp(cmd, "recovery") == 0 || strcmp(cmd, "factory_reset") == 0)
             reboot_reason = MESON_FACTORY_RESET_REBOOT;
@@ -61,6 +63,7 @@
             reboot_reason = MESON_USB_BURNER_REBOOT;
 	}
     aml_write_reg32(P_AO_RTI_STATUS_REG1, reboot_reason);
+    printk("meson common restart \n");
     printk("reboot_reason(0x%x) = 0x%x\n", P_AO_RTI_STATUS_REG1, aml_read_reg32(P_AO_RTI_STATUS_REG1));
 	arch_reset(mode,cmd);
 }
@@ -71,8 +74,8 @@
 
 void meson_power_off(void)
 {
-	printk("meson power off \n");
-	meson_common_restart('h',"charging_reboot");
+        printk("meson power off \n");
+        meson_common_restart('h', "poweroff");
 }
 
 void meson_power_idle(void)
