diff --git a/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp b/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp
index 28368fe..1525968 100644
--- a/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp
@@ -65,7 +65,7 @@ void CEGLNativeTypeAmlogic::Initialize()
   aml_permissions();
   aml_cpufreq_min(true);
   aml_cpufreq_max(true);
-  return;
+  DisableFreeScale();
 }
 void CEGLNativeTypeAmlogic::Destroy()
 {
@@ -315,17 +316,18 @@ void CEGLNativeTypeAmlogic::DisableFreeScale()
   // turn off frame buffer freescale
   aml_set_sysfs_int("/sys/class/graphics/fb0/free_scale", 0);
   aml_set_sysfs_int("/sys/class/graphics/fb1/free_scale", 0);
-  aml_set_sysfs_str("/sys/class/graphics/fb0/free_scale_axis", "0 0 1279 719");
+  // aml_set_sysfs_str("/sys/class/graphics/fb0/free_scale_axis", "0 0 1279 719");
 
-  aml_set_sysfs_int("/sys/class/ppmgr/ppscaler", 0);
-  aml_set_sysfs_int("/sys/class/video/disable_video", 0);
-  // now default video display to off
-  aml_set_sysfs_int("/sys/class/video/disable_video", 1);
+  // aml_set_sysfs_int("/sys/class/ppmgr/ppscaler", 0);
+  // aml_set_sysfs_int("/sys/class/video/disable_video", 0);
+  //  // now default video display to off
+  // aml_set_sysfs_int("/sys/class/video/disable_video", 1);
 
-  // revert display axis
-  int fd0;
-  std::string framebuffer = "/dev/" + m_framebuffer_name;
+  //  // revert display axis
+  // int fd0;
+  // std::string framebuffer = "/dev/" + m_framebuffer_name;
 
+/*
   if ((fd0 = open(framebuffer.c_str(), O_RDWR)) >= 0)
   {
     struct fb_var_screeninfo vinfo;
@@ -337,6 +339,7 @@ void CEGLNativeTypeAmlogic::DisableFreeScale()
     }
     close(fd0);
   }
+*/
 }
 
 void CEGLNativeTypeAmlogic::SetFramebufferResolution(const RESOLUTION_INFO &res) const
