diff -Nru kodi-14-f4576be.orig/xbmc/utils/CPUInfo.cpp kodi-14-f4576be/xbmc/utils/CPUInfo.cpp
--- kodi-14-f4576be.orig/xbmc/utils/CPUInfo.cpp	2014-12-09 22:29:43.000000000 +0300
+++ kodi-14-f4576be/xbmc/utils/CPUInfo.cpp	2014-12-11 21:47:28.603633663 +0300
@@ -262,7 +262,7 @@
     m_fProcTemperature = fopen("/proc/acpi/thermal_zone/TZ0/temperature", "r");
   // read from the new location of the temperature data on new kernels, 2.6.39, 3.0 etc
   if (m_fProcTemperature == NULL)   
-    m_fProcTemperature = fopen("/sys/class/hwmon/hwmon0/temp1_input", "r");
+    m_fProcTemperature = fopen("/sys/class/hwmon/hwmon1/temp1_input", "r");
   if (m_fProcTemperature == NULL)   
     m_fProcTemperature = fopen("/sys/class/thermal/thermal_zone0/temp", "r");  // On Raspberry PIs
 
