diff -Nru kodi-14-f4576be.orig/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp kodi-14-f4576be/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
--- kodi-14-f4576be.orig/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp	2014-12-09 22:29:46.000000000 +0300
+++ kodi-14-f4576be/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp	2014-12-15 14:30:25.329034786 +0300
@@ -26,6 +26,7 @@
 #include "AMLCodec.h"
 #include "utils/BitstreamConverter.h"
 #include "utils/log.h"
+#include "settings/AdvancedSettings.h"
 
 #define __MODULE_NAME__ "DVDVideoCodecAmlogic"
 
@@ -69,6 +70,12 @@
     case AV_CODEC_ID_MPEG1VIDEO:
     case AV_CODEC_ID_MPEG2VIDEO:
     case AV_CODEC_ID_MPEG2VIDEO_XVMC:
+       if (g_advancedSettings.m_AMLcodecUseMpeg2codec != 0 && m_hints.width < 1280)
+         {
+           CLog::Log(LOGDEBUG, "use SW decoding for MPEG2 codec");
+           return false;
+           break;
+         }
       m_mpeg2_sequence_pts = 0;
       m_mpeg2_sequence = new mpeg2_sequence;
       m_mpeg2_sequence->width  = m_hints.width;
diff -Nru kodi-14-f4576be.orig/xbmc/settings/AdvancedSettings.cpp kodi-14-f4576be/xbmc/settings/AdvancedSettings.cpp
--- kodi-14-f4576be.orig/xbmc/settings/AdvancedSettings.cpp	2014-12-09 22:29:45.000000000 +0300
+++ kodi-14-f4576be/xbmc/settings/AdvancedSettings.cpp	2014-12-15 14:02:43.080479640 +0300
@@ -175,6 +175,7 @@
   m_stagefrightConfig.useMPEG2codec = -1;
   m_stagefrightConfig.useSwRenderer = false;
   m_stagefrightConfig.useInputDTS = false;
+  m_AMLcodecUseMpeg2codec = false;
 
   m_mediacodecForceSoftwareRendring = false;
 
@@ -596,6 +597,7 @@
     XMLUtils::GetBoolean(pElement,"vdpauInvTelecine",m_videoVDPAUtelecine);
     XMLUtils::GetBoolean(pElement,"vdpauHDdeintSkipChroma",m_videoVDPAUdeintSkipChromaHD);
     XMLUtils::GetBoolean(pElement,"useffmpegvda", m_useFfmpegVda);
+    XMLUtils::GetBoolean(pElement, "amluseswmpeg2", m_AMLcodecUseMpeg2codec);
 
     TiXmlElement* pStagefrightElem = pElement->FirstChildElement("stagefright");
     if (pStagefrightElem)
diff -Nru kodi-14-f4576be.orig/xbmc/settings/AdvancedSettings.h kodi-14-f4576be/xbmc/settings/AdvancedSettings.h
--- kodi-14-f4576be.orig/xbmc/settings/AdvancedSettings.h	2014-12-09 22:29:45.000000000 +0300
+++ kodi-14-f4576be/xbmc/settings/AdvancedSettings.h	2014-12-15 13:26:42.953586132 +0300
@@ -197,6 +197,7 @@
     int  m_videoFpsDetect;
     int  m_videoBusyDialogDelay_ms;
     StagefrightConfig m_stagefrightConfig;
+    bool m_AMLcodecUseMpeg2codec;
     bool m_mediacodecForceSoftwareRendring;
 
     CStdString m_videoDefaultPlayer;
