diff -Nru linux-amlogic-3.10.40.orig/drivers/amlogic/wifi/rtl8xxx_EU/include/autoconf.h linux-amlogic-3.10.40/drivers/amlogic/wifi/rtl8xxx_EU/include/autoconf.h
--- linux-amlogic-3.10.40.orig/drivers/amlogic/wifi/rtl8xxx_EU/include/autoconf.h	2014-11-24 04:54:32.000000000 +0300
+++ linux-amlogic-3.10.40/drivers/amlogic/wifi/rtl8xxx_EU/include/autoconf.h	2014-11-25 09:51:40.000000000 +0300
@@ -107,7 +107,7 @@
 	#endif
 
 
-	//#define CONFIG_CONCURRENT_MODE 
+	#define CONFIG_CONCURRENT_MODE 
 	#ifdef CONFIG_CONCURRENT_MODE
 		//#define CONFIG_HWPORT_SWAP				//Port0->Sec , Port1 -> Pri
 		#define CONFIG_RUNTIME_PORT_SWITCH
diff -Nru linux-amlogic-3.10.40.orig/drivers/amlogic/wifi/rtl8xxx_EU_MP/include/autoconf.h linux-amlogic-3.10.40/drivers/amlogic/wifi/rtl8xxx_EU_MP/include/autoconf.h
--- linux-amlogic-3.10.40.orig/drivers/amlogic/wifi/rtl8xxx_EU_MP/include/autoconf.h	2014-11-24 04:54:32.000000000 +0300
+++ linux-amlogic-3.10.40/drivers/amlogic/wifi/rtl8xxx_EU_MP/include/autoconf.h	2014-11-25 09:51:40.000000000 +0300
@@ -109,7 +109,7 @@
 	#endif
 
 
-	//#define CONFIG_CONCURRENT_MODE 
+	#define CONFIG_CONCURRENT_MODE 
 	#ifdef CONFIG_CONCURRENT_MODE
 		//#define CONFIG_HWPORT_SWAP				//Port0->Sec , Port1 -> Pri
 		#define CONFIG_TSF_RESET_OFFLOAD 			// For 2 PORT TSF SYNC.
diff -Nru linux-amlogic-3.10.40.orig/drivers/amlogic/wifi/rtl8xxx_EU_MP/Kconfig linux-amlogic-3.10.40/drivers/amlogic/wifi/rtl8xxx_EU_MP/Kconfig
--- linux-amlogic-3.10.40.orig/drivers/amlogic/wifi/rtl8xxx_EU_MP/Kconfig	2014-11-24 04:54:32.000000000 +0300
+++ linux-amlogic-3.10.40/drivers/amlogic/wifi/rtl8xxx_EU_MP/Kconfig	2014-11-25 09:51:40.000000000 +0300
@@ -1,5 +1,5 @@
 config RTL8188EU_MP
-	tristate "Realtek 8188EU-MP USB WiFi"
+	tristate "Realtek 8188E USB WiFi"
 	depends on USB
 	---help---
 	  Help message of RTL8188EU
