diff -Nru linux-amlogic-3.10-fa4c242.orig/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd linux-amlogic-3.10-fa4c242/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd
--- linux-amlogic-3.10-fa4c242.orig/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd	2015-01-22 23:04:54.000000000 +0300
+++ linux-amlogic-3.10-fa4c242/arch/arm/boot/dts/amlogic/meson8_k200a_1G.dtd	2015-01-30 01:57:49.370266458 +0300
@@ -1332,11 +1332,15 @@
 //$$ L2 PROP_U32 = "vendor_id"
 //$$ L2 PROP_STR = "product_desc"
 //$$ L2 PROP_STR = "cec_osd_string"        
+//$$ L2 PROP_U32 = "cec_config"
+//$$ L2 PROP_U32 = "ao_cec"
         vend_data: vend_data{
             vendor_name = "Amlogic";           /* Max Chars: 8     */
             vendor_id = <0x000000>;                 /* Refer to http://standards.ieee.org/develop/regauth/oui/oui.txt   */
             product_desc = "M8 MBox SKTv1";        /* Max Chars: 16    */
             cec_osd_string = "Amlogic MBox";        /* Max Chars: 14    */
+            cec_config = <0x00000f>;           /* 4 bytes: use to control cec switch on/off; distinguish between Mbox and Tablet. <0x00000f>:Mbox; <0x000000> or no this value:Tablet*/
+            ao_cec = <0x000001>;               /* switch between ao-cec and ee-cec:: 1: ao-cec; 0: ee-cec */
         };
         
 //$$ MATCH "HDMI_pwr-ctrl_match" = <&pwr_ctrl>
